vocabulary-care-pathways-questions/api/src/main/java/org/phenotips/vocabulary/internal/solr/CarePathwaysQuestionOntology.java

```diff
@@ -0,0 +1,350 @@
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.text.MessageFormat;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrQuery.ORDER;
+import org.apache.solr.client.solrj.util.ClientUtils;
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.params.DisMaxParams;
+import org.apache.solr.common.params.SpellingParams;
+
+/**
+ * Provides access to the Care Pathways questions vocabulary. The vocabulary prefix is {@code CPQ}.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component
+@Named("care-pathways-questions")
+@Singleton
+public class CarePathwaysQuestionOntology extends AbstractCSVSolrVocabulary
+{
+    /** The category for care pathways question-related terms. */
+    private static final String QUESTION_CATEGORY = "survey-question";
+
+    /** The standard prefix for the vocabulary terms. */
+    private static final String STANDARD_PREFIX = "CPQ";
+
+    /** A colon. */
+    private static final String COLON = ":";
+
+    /** The label for the field storing ID for term. */
+    private static final String ID = "id";
+
+    /** The label for the field storing the question string. */
+    private static final String NAME = "name";
+
+    /** The label for the field storing the question group. */
+    private static final String TERM_GROUP = "term_group";
+
+    /** For determining if a query is a an id. */
+    private static final Pattern ID_PATTERN = Pattern.compile("^CPQ:[0-9]+$", Pattern.CASE_INSENSITIVE);
+
+    @Override
+    protected int getSolrDocsPerBatch()
+    {
+        return 500000;
+    }
+
+    @Override
+    protected String getCoreName()
+    {
+        return getIdentifier();
+    }
+
+    @Override
+    public String getIdentifier()
+    {
+        return "care-pathways-questions";
+    }
+
+    @Override
+    public String getName()
+    {
+        return "Care pathways survey questions";
+    }
+
+    @Override
+    public Set<String> getAliases()
+    {
+        Set<String> aliases = new HashSet<>();
+        aliases.add(getIdentifier());
+        aliases.add(STANDARD_PREFIX);
+        return aliases;
+    }
+
+    @Override
+    public Collection<String> getSupportedCategories()
+    {
+        return Collections.singletonList(QUESTION_CATEGORY);
+    }
+
+    @Override
+    public String getDefaultSourceLocation()
+    {
+        return StringUtils.EMPTY;
```

Comment:
Same comment as on #4, use the classloader to get the location of a bundled copy of the file.

Reply:
Fixed.